### PR TITLE
removed subkey from authenticator within krb5Token

### PR DESCRIPTION
**Pull request by Shastick**
The sequence number has already been generated in `types.NewAuthenticator`, and a subkey is not strictly required.

For instances where a subkey is required and depending on the use case (ie, when the Authenticator is generated through a call to `NewNegTokenInitKrb5`), it will need to be propagated back to the caller before the authenticator is serialized.

Remarks:
  - Added a test case to check the authenticator for both cases
  - `NewAuthenticator ` now has a useless parameter (`keyType int`). Options: add a new function with another name and deprecate this one?
  - Other option: leave `NewAuthenticator` as is and create another function that does not generate subkeys

On testing:
 - all package tests seem to pass.
 - the current code allows me to do a kerberos authentication handshake with an hadoop namenode. I don't know if everything is as it should be per RFCs, but in my current use case it works.
 - I have no clue if serialising the Authenticator struct without a subkey yields the expected results in other situations (ie, SPNEGO or anything else).

PS: at least another PR is in the pipeline, I prefer to keep things small and simple :)

### CODE REVIEW

File: gssapi/krb5Token_test.go [hunk 1]
```diff
@@ -43,17 +43,20 @@ func TestMechToken_newAuthenticatorChksum(t *testing.T) {
 	assert.Equal(t, b, cb, "SPNEGO Authenticator checksum not as expected")
 }
 
-func TestMechToken_newAuthenticator(t *testing.T) {
+// Test with explicit subkey generation.
+func TestMechToken_newAuthenticatorWithSubkeyGeneration(t *testing.T) {
 	creds := credentials.NewCredentials("hftsai", testdata.TEST_REALM)
 	creds.CName.NameString = testdata.TEST_PRINCIPALNAME_NAMESTRING
 	etypeID := 18
+	keyLen := 32 // etypeID 18 refers to AES255 -> 32 bytes key
```
Comment (Shastick) [inline gssapi/krb5Token_test.go]: Small typo in comment here. Will fix later.

Comment (jcmturner) [reply gssapi/krb5Token_test.go]: Thanks, please fix the typo so that I can merge.

**Comment by jcmturner** [comment]
relates to issue #65 comment https://github.com/jcmturner/gokrb5/issues/65#issuecomment-358014204

File: gssapi/krb5Token.go [hunk 2]
```diff
@@ -155,11 +154,6 @@ func NewAuthenticator(creds credentials.Credentials, keyType int, flags []int) (
 	if err != nil {
 		return auth, krberror.Errorf(err, krberror.KRBMsgError, "error generating new authenticator")
 	}
-	etype, err := crypto.GetEtype(keyType)
-	if err != nil {
-		return auth, krberror.Errorf(err, krberror.KRBMsgError, "error getting etype for authenticator")
-	}
-	auth.GenerateSeqNumberAndSubKey(keyType, etype.GetKeyByteSize())
```
Comment (jcmturner) [inline gssapi/krb5Token.go]: As you mention in your PR comment the NewAuthenticator method now has a pointless keyType argument. I think we should update the comment on this to be something like: ```// NewAuthenticator (DEPRECATED - this method will be updated in future versions to remove the obsolete keyType argument and may be made private to the gssapi package) creates a new kerberos authenticator for kerberos MechToken```

**Comment by jcmturner** [comment]
@Shastick Thanks very much for your first PR. I recently added [contribution guidelines for PRs](https://github.com/jcmturner/gokrb5/blob/master/CONTRIBUTING.md#code-contribution-ground-rules). Please take a look at these for future PRs.

I have updated the title to fit the guidelines and added some review comments. Please can you take a look at these and commit the required updates. Thanks.

I had specified that PR comments should be wrapped to 72 characters but I'm reconsidering this as your comments were very helpful and I'm keen to encourage this over formatting.

**Comment by Shastick** [comment]
Thanks for the feedback on this and the other PR.

I'll keep the verbosity for the commit message next time, unless explanations in the PR are really required. (I'm relatively new to the git PR workflow and did not realize that PR message != commit message). 

Some things I mentioned in the PR could probably have been discussed in a dedicated issue (as the guidelines suggest I should have done ;)). I just slipped back in a habit of having one/few commit per feature with enough explanations so a look at a single entry in `git log` has all the info one needs.

**Comment by jcmturner** [comment]
I appreciate the effort you took to explain the PR. I don't want to discourage that!
As you say an issue may be the better place for more verbose discussions. This is the first open source project I'm managing so we're both learning :-)

**Pull request merged**